File: 백준/Silver/2839. 설탕 배달/설탕 배달.cc
#include <iostream>
using namespace std;

int N, M;

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

	cin >> N;
	int ans = 0;
	int Five = 0, Three = 0;
	Five = N / 5;
	
	int T = N % 5;
	Three += T / 3;
	T = T % 3;
	while (T--)
	{
		if (Five <= 0)
		{
			ans = -1;
			break;
		}
		Five--;
		Three += 2;
	}
	if (ans != -1) cout << Five + Three;
	else cout << -1;
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.*;
import java.util.*;
public class Main
{
	public static void main(String[] args) throws Exception {
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	    int N = Integer.parseInt(br.readLine());
	    
	    int Five = 0;
	    int Three = 0;
	    boolean ans = false;
	    Five += N / 5;
	    N %= 5;
	    Three += N /3;
	    N %= 3;
	    
	    while (N > 0)
	    {
	        if (Five <= 0)
	        {
	            ans = true;
	            break;
	        }
	        Five--;
	        Three += 2;
	        N--;
	    }
	    if (ans) System.out.println(-1);
	    else System.out.println(Five + Three);
	}
}